pkg/keymanagementprovider/keymanagementprovider.go

```diff
@@ -111,7 +115,7 @@ func SetCertificatesInMap(resource string, certs map[KMPMapKey][]*x509.Certifica
 
 // GetCertificatesFromMap gets the certificates from the map and returns an empty map of certificate arrays if not found
 func GetCertificatesFromMap(ctx context.Context, resource string) map[KMPMapKey][]*x509.Certificate {
-	if !isCompatibleNamespace(ctx, resource) {
+	if !hasAccessToProvider(ctx, resource) {
```

Comment:
I'm curious how inaccessible certs bubbles up to users? If a user accidentally configured a reference to a KMP that this cluster resource does not have access to, how would they know?

Reply:
I see that we have an error thrown when `GetKeysFromMap` return `false` in the cosign trust policy  scenario. And throws an error about `no certificates found` in the notation truststore. However, maybe we can consider improving error message to distinguish when it's a permission issue instead of a KMP with no certificates/keys that could be fetched?

Reply:
Good call out! I'll make GetKeysFromMap and GetCertificatesFromMap returns result and an error so that we can catch up and surface out it.